src/interceptors/xhrInterceptor.js

```diff
@@ -72,7 +72,16 @@ const fakeService = helpers => class XMLHttpRequestInterceptor {
       //TODO: should 'this.response' be the response string or the response json?
       this.responseText = this.response = responseString;
       
-      return successCallback.call(this);
+      const requestDelay = helpers.getDelay();
+      if (!requestDelay) {
+        // for backwards compatibility if there is no requestDelay ot it 0 return callback without setTimeout
+        return successCallback.call(this);
+      } else {
+        return setTimeout(
+          () => successCallback.call(this),
+          requestDelay
+        );
+      }      
```

Comment:
I'll say just remove the conditional and always use `setTimeout` since the default value of `requestDelay` is zero:

```javascript
const requestDelay = helpers.getDelay();

return setTimeout(
  () => successCallback.call(this),
  requestDelay
);
```